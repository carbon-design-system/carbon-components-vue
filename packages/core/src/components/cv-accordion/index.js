import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
