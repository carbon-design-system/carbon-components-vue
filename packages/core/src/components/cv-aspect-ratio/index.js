import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
