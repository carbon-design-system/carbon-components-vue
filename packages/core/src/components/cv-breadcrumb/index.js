import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
