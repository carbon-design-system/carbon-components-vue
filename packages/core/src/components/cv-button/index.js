import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
