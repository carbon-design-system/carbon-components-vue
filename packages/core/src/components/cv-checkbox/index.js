import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
