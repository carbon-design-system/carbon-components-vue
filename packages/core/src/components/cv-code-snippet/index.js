import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
