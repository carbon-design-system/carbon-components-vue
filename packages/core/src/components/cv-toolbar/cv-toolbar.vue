<template>
  <div :class="`${carbonPrefix}--toolbar`" data-toolbar>
    <slot></slot>
  </div>
</template>

<script>
import { carbonPrefixMixin } from '../../mixins';
export default {
  name: 'CvToolbar',
  mixins: [carbonPrefixMixin],
};
</script>
