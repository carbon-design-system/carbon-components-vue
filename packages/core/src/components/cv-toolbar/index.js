import CvToolbar from './cv-toolbar';
import CvToolbarTitle from './cv-toolbar-title';
import CvToolbarSearch from './cv-toolbar-search';
import CvToolbarOption from './cv-toolbar-option';
import CvToolbarDivider from './cv-toolbar-divider';
export { CvToolbar, CvToolbarTitle, CvToolbarSearch, CvToolbarOption, CvToolbarDivider };
