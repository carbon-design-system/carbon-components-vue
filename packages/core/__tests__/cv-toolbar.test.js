import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow, trigger, mount } = awaitNextTick;
import { CvToolbar, CvToolbarTitle, CvToolbarSearch, CvToolbarOption, CvToolbarDivider } from '@/components/cv-toolbar';
import { settings as carbonSettings } from 'carbon-components';

const carbonPrefix = carbonSettings.prefix;

describe('CvToolbar', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render without slot correctly', async () => {
    const propsData = { id: 'test-1' };
    const wrapper = await shallow(CvToolbar, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render with slot correctly', async () => {
    const propsData = { id: 'test-1' };
    const wrapper = await shallow(CvToolbar, {
      slots: {
        default: '<div>default test slot</div>',
      },
      propsData,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
});

describe('CvToolbarTitle', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvToolbarTitle, ['title'], String);
  testComponent.propsAreRequired(CvToolbarTitle, ['title']);

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render correctly', async () => {
    const propsData = { id: 'test-1', title: 'Toolbar test title' };
    const wrapper = await shallow(CvToolbarTitle, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
});

describe('CvToolbarSearch', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvToolbarSearch, ['value'], String);

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render correctly', async () => {
    const propsData = { id: 'test-1', value: 'check-1' };
    const wrapper = await shallow(CvToolbarSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
  it('should add active css class on focus and remove on blur', async () => {
    const propsData = { id: 'test-1', value: 'check-1' };
    const wrapper = await mount(CvToolbarSearch, { propsData });

    await trigger(wrapper.find(`.${carbonPrefix}--toolbar-search__btn`), 'click');
    expect(wrapper.classes()).toContain(`${carbonPrefix}--toolbar-search--active`);

    await trigger(wrapper.find(`.${carbonPrefix}--toolbar-search__btn`), 'blur');
    expect(wrapper.classes()).not.toContain(`${carbonPrefix}--toolbar-search--active`);
  });
});

describe('CvToolbarOption', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render without slot correctly', async () => {
    const propsData = { id: 'test-1' };
    const wrapper = await shallow(CvToolbarOption, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render with slot correctly', async () => {
    const propsData = { id: 'test-1' };
    const wrapper = await shallow(CvToolbarOption, {
      slots: {
        default: '<div>default test slot</div>',
      },
      propsData,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
  it('parent should emit event on esc keydown', async () => {
    const propsData = { id: 'test-1' };
    const wrapper = await mount(CvToolbar, {
      slots: {
        default: CvToolbarOption,
      },
      propsData,
    });

    await trigger(wrapper.find(CvToolbarOption), 'keydown.esc');
    expect(wrapper.emitted('cv:close')).toBeTruthy();
  });
});

describe('CvToolbarDivider', () => {
  // ***************
  // PROP CHECKS
  // ***************

  // ***************
  // SNAPSHOT TESTS
  // ***************

  it('should render correctly', async () => {
    const propsData = { id: 'test-1' };
    const wrapper = await shallow(CvToolbarDivider, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************
});
