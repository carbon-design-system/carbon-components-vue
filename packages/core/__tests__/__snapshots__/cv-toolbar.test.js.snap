// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvToolbar should render with slot correctly 1`] = `
<div data-toolbar="" class="bx--toolbar" id="test-1">
  <div>default test slot</div>
</div>
`;

exports[`CvToolbar should render without slot correctly 1`] = `<div data-toolbar="" class="bx--toolbar" id="test-1"></div>`;

exports[`CvToolbarDivider should render correctly 1`] = `<hr class="cv-toolbar-divider bx--toolbar-menu__divider" id="test-1">`;

exports[`CvToolbarOption should render with slot correctly 1`] = `
<li class="cv-toolbar-option bx--toolbar-menu__option" id="test-1">
  <div>default test slot</div>
</li>
`;

exports[`CvToolbarOption should render without slot correctly 1`] = `<li class="cv-toolbar-option bx--toolbar-menu__option" id="test-1"></li>`;

exports[`CvToolbarSearch should render correctly 1`] = `<cv-search-stub id="test-1" cleararialabel="Clear search input" kind="toolbar" small="true" value="check-1" placeholder="Search" toolbararialabel="Toolbar search" class="cv-toolbar-search"></cv-search-stub>`;

exports[`CvToolbarTitle should render correctly 1`] = `<li class="cv-toolbar-title bx--toolbar-menu__title" id="test-1">Toolbar test title</li>`;
